rest/changes_api_test.go

```diff
@@ -1042,6 +1042,150 @@ func TestOneShotChangesWithExplicitDocIds(t *testing.T) {
+	response = rt.SendAdminRequest("PUT", "/db/doc_conflict?new_edits=false", newEdits_conflict)
+
+	// Resolved conflict
+	revid, err = updateTestDoc(rt, "doc_resolved_conflict", "", `{"type": "resolved_conflict", "channels":["alpha"]}`)
+	_, err = updateTestDoc(rt, "doc_resolved_conflict", revid, `{"type": "resolved_conflict", "channels":["alpha"]}`)
+	newEdits_conflict = fmt.Sprintf(`{"type": "resolved_conflict", "channels":["alpha"]},
+                   "_revisions": {"start": 2, "ids": ["conflicting_rev", "%s"]}}`, revid)
+	response = rt.SendAdminRequest("PUT", "/db/doc_resolved_conflict?new_edits=false", newEdits_conflict)
+	response = rt.SendAdminRequest("DELETE", "/db/doc_resolved_conflict?rev=2-conflicting_rev", "")
+
+	expectedResults := make([]string, 9)
+	expectedResults[0] = `{"seq":1,"id":"_user/user1","changes":[]}`
+	expectedResults[1] = `{"seq":2,"id":"doc_active","doc":{"_id":"doc_active","_rev":"1-d59fda97ac4849f6a754fbcf4b522980","channels":["alpha"],"type":"active"},"changes":[{"rev":"1-d59fda97ac4849f6a754fbcf4b522980"}]}`
+	expectedResults[2] = `{"seq":4,"id":"doc_tombstone","deleted":true,"removed":["alpha"],"doc":{"_deleted":true,"_id":"doc_tombstone","_rev":"2-5bd8eb422f30e8d455940672e9e76549"},"changes":[{"rev":"2-5bd8eb422f30e8d455940672e9e76549"}]}`
+	expectedResults[3] = `{"seq":6,"id":"doc_removed","removed":["alpha"],"doc":{"_id":"doc_removed","_removed":true,"_rev":"2-d15cb77d1dbe1cc06d27310de5b75914"},"changes":[{"rev":"2-d15cb77d1dbe1cc06d27310de5b75914"}]}`
+	expectedResults[4] = `{"seq":10,"id":"doc_pruned","removed":["alpha"],"doc":{"_id":"doc_pruned","_removed":true,"_rev":"2-5afcb73bd3eb50615470e3ba54b80f00"},"changes":[{"rev":"2-5afcb73bd3eb50615470e3ba54b80f00"}]}`
+	expectedResults[5] = `{"seq":16,"id":"doc_attachment","doc":{"_attachments":{"attach1":{"content_type":"text/plain","digest":"sha1-nq0xWBV2IEkkpY3ng+PEtFnCcVY=","length":30,"revpos":2,"stub":true}},"_id":"doc_attachment","_rev":"2-0db6aecd6b91981e7f97c95ca64b5019","channels":["alpha"],"type":"attachments"},"changes":[{"rev":"2-0db6aecd6b91981e7f97c95ca64b5019"}]}`
+	expectedResults[6] = `{"seq":17,"id":"doc_large_numbers","doc":{"_id":"doc_large_numbers","_rev":"1-2721633d9000e606e9c642e98f2f5ae7","channels":["alpha"],"largefloat":1234567890.1234,"largeint":1234567890,"type":"large_numbers"},"changes":[{"rev":"1-2721633d9000e606e9c642e98f2f5ae7"}]}`
+	expectedResults[7] = `{"seq":19,"id":"doc_conflict","doc":{"_id":"doc_conflict","_rev":"2-869a7167ccbad634753105568055bd61","channels":["alpha"],"type":"conflict"},"changes":[{"rev":"2-869a7167ccbad634753105568055bd61"}]}`
+	expectedResults[8] = `{"seq":21,"id":"doc_resolved_conflict","doc":{"_id":"doc_resolved_conflict","_rev":"2-251ba04e5889887152df5e7a350745b4","channels":["alpha"],"type":"resolved_conflict"},"changes":[{"rev":"2-251ba04e5889887152df5e7a350745b4"}]}`
+	changesResponse := rt.Send(requestByUser("GET", "/db/_changes?include_docs=true", "", "user1"))
```

Comment:
What about repeating this changes request to cover all of the scenarios?

- options.IncludeDocs && includeConflicts
- includeConflicts
- options.IncludeDocs

Reply:
The motivation of this particular test was to ensure that the include_docs output remains consistent after making the switch to using the rev cache.

Each of the options you listed will have a different expected result, so I don't know if it really makes sense in the context of this test.  We don't seem to have test coverage elsewhere, though, so I'll take a look to see if it makes sense here.

Reply:
I tried kicking off a coverage test in http://uberjenkins.sc.couchbase.com/view/Build/job/sync-gateway-coverage/17/, but it hit a sporadic error.  

Ok great.  I do think it's probably a good opportunity to add extra test coverage while we're in this code.